### ISSUE TITLE
Using rawCommand to send module added commands fails

### ISSUE THREAD

**Issue by rvencu** [phpredis/phpredis#1826]
### Expected behaviour
$redis->rawCommand("set", $key . $value);
$redis->rawCommand("TRIE.INSERT", $key . $value);
both should work when these commands work OK from redis-cli

### Actual behaviour
$redis->rawCommand("set", $key, $value); works nice
$redis->rawCommand("TRIE.INSERT", $key, $value); does not work
errors start with `socket error on read socket` followed by `Connection lost` followed by `Redis server 127.0.0.1:6379 went away`

### I'm seeing this behaviour on
- OS: centos7
- Redis: 5.0.0
- PHP: 7.3.24
- phpredis: 5.3.1

### Steps to reproduce, backtrace or example script
as above with the trie module installed https://github.com/cmsc22000-project-2018/redis-tries

### I've checked
- [x] There is no similar issue from other users
- [ ] Issue isn't fixed in `develop` branch

**Comment by michael-grunder** [comment]
Are you sure the Redis instance didn't crash?

I installed the module locally and had no problem interacting with its commands using `rawCommand`.  The errors you show is what I would expect if Redis crashed.

I haven't tried with the exact version of PHP you're using (5.3.1) so I'll try that over the weekend, but `rawCommand` has no custom logic at all so it doesn't make sense why it would break.

**Comment by rvencu** [comment]
While getting these errors, the redis-cli was working. But the Redis Cache Object plugin for WordPress also reported disconnection

I am not sure if it allowed to perform multiple connections through phpredis from the same session to the same Redis server though...

I will perform more tests though to check how can Redis server crash including log check

**Comment by rvencu** [comment]
Yes, these commands completely crash Redis server https://github.com/redis/redis/issues/7658

**Comment by michael-grunder** [comment]
I'd open your issue over on the [module repo](https://github.com/cmsc22000-project-2018/redis-tries/issues) since it's crashing in that code and isn't likely a problem with Redis itself.

**Comment by rvencu** [comment]
Thanks, copied it there https://github.com/cmsc22000-project-2018/redis-tries/issues/113

**Comment by michael-grunder** [comment]
Cool, I hope the issue gets sorted out.  I'm going to close this issue since it's not a bug in PhpRedis.

Edit:  Looks like the other repo was just a school project.  If you need a trie you might want to see if someone else has an implementation.  [Here's one](https://github.com/arashThr/trie-redis-module) although it doesn't seem widely used.

What functionality do you need?  [RedisLabs](https://redislabs.com/redis-enterprise/modules/) has quite a few high-quality modules.

**Comment by rvencu** [comment]
Thanks. I need to add words and phrases into a dictionary and then search for their existence as fast as possible. I implemented a trie class in PHP and stuffed into a trie object some 1 million terms. I can filter out from a list of terms what exists and what not pretty quickly but the generation of the trie object at every call takes about 3.4 seconds. This time is a waste at every page load. I used to keep the trie object persistent in APC cache (it is a good solution) but then I believe I need to update the technology and not only keep it persistent in REDIS but also benefit of the REDIS speed over PHP speed while searching a potentially even larger collection of terms.

Basically I only need 2 functions: insert and contains. No need for autocomplete at this stage, however, autocomplete could be handy at future steps of my project.

I'll check other solutions of course.

**Comment by rvencu** [comment]
In fact, the trie object I finally get is 260MB in size. I tested with all available caching mechanisms today: APCu, Memcached, Redis by trying to save and retrieve the entire object content. In most cases, it takes longer time than loading from file and unserialize the content. So I guess the Redis trie structure is actually a must now...

**Comment by michael-grunder** [comment]
Why not just use a regular `set` or `zset`?

You could also take a look at [RediSearch](https://oss.redislabs.com/redisearch/) which I think can also give you autocomplete.

**Comment by rvencu** [comment]
Regular set of the entire trie object works fine but the get still takes 3 seconds to rebuild the object in PHP session.

Thanks for the suggestion about RediSearch which shows big potential to improve on the solution. I will follow it up and see where it can take me.
